### PR TITLE
Configure domain

### LINKED ISSUE #121: Change dns to *.dev
How can I use *.dev instead of *.docker?

### PR DESCRIPTION
**Pull request by ryansch**
Fixes on top of #124.  I kept the commits so that @adrianfalleiro gets credit.

Fixes #121 
Closes #124

### CODE REVIEW

**Comment by ryansch** [comment]
The dinghy-http-proxy does not respect this setting.

File: cli/dinghy/dnsmasq.rb [hunk 1]
```diff
@@ -5,12 +5,18 @@
...
   RESOLVER_DIR = Pathname("/etc/resolver")
-  RESOLVER_FILE = RESOLVER_DIR.join("docker")
 
-  attr_reader :machine
+  attr_reader :machine, :resolver_file, :dinghy_domain
 
   def initialize(machine)
     @machine = machine
+    self.dinghy_domain = "docker"
+    @resolver_file = RESOLVER_DIR.join(@dinghy_domain)
```
Comment (codekitchen) [inline cli/dinghy/dnsmasq.rb]: (from the original review) this line can be removed

**Comment by codekitchen** [comment]
If you explicitly set a VIRTUAL_HOST ending in the same TLD that you set dinghy_domain to, then the http proxy should work right? The hostnames auto-generated based on the docker-compose tags won't, though. That'd be pretty easy to fix by passing dinghy_domain as an env var into the container and having the template use that.

**Comment by ryansch** [comment]
Sorry that comment was about the auto generated hostnames.

File: cli/dinghy/dnsmasq.rb (hunk 1 above)
Comment (ryansch) [inline cli/dinghy/dnsmasq.rb]: Yeah I just now picked up on the fact that we're using the setter for `dinghy_domain` instead of the ivar.

Comment (codekitchen) [inline cli/dinghy/dnsmasq.rb]: No worries, thanks for fixing up this PR

**Comment by codekitchen** [comment]
I'll tackle the auto generated hostnames in a separate commit, it's an unsupported "bonus" feature right now anyway. Though I'm leaning toward making it an official feature, it's turned out to be really nice and it's even nicer when the TLD can change between different people's computers.

**Comment by ryansch** [comment]
Roger that.

**Comment by codekitchen** [comment]
Merged, thanks again

**Comment by adrianfalleiro** [comment]
Thanks for picking this up @ryansch. Looks great :)

**Pull request merged**